components/ChatApp.tsx

```diff
@@ -106,60 +138,100 @@ export class ChatApp extends React.Component<
+                                                        ...prevState.onlineMembers,
+                                                        member.identity,
+                                                    ],
+                                                }));
                                             } else {
-                                                this.setState(
-                                                    (prevState, props) => ({
-                                                        offlineMembers: [
-                                                            ...prevState.offlineMembers,
-                                                            member.identity,
-                                                        ],
-                                                    }),
-                                                );
+                                                this.setState((prevState) => ({
+                                                    offlineMembers: [
+                                                        ...prevState.offlineMembers,
+                                                        member.identity,
+                                                    ],
+                                                }));
                                             }
                                         });
                                     }}
                                 >
-                                    {this.state.channels.map((channel, i) => (
-                                        <li key={i}>
-                                            <button
-                                                type="submit"
-                                                name={channel}
-                                                onClick={(event: any) => {
-                                                    this.setState({
-                                                        joinChannel:
-                                                            event.target.value,
-                                                    });
-                                                }}
-                                                value={channel}
-                                            >
-                                                {channel}
-                                            </button>
-                                        </li>
-                                    ))}
+                                    {this.state.channels
+                                        .sort(
+                                            (a, b) =>
+                                                b.channel.dateCreated -
+                                                a.channel.dateCreated,
+                                        )
+                                        .map((channel) => (
+                                            <li key={channel.channel.sid}>
```

Comment:
having a var called channel and using channel.channel probably means channel var should not be called channel, can you find a better name?